### PR TITLE
PR: Allow tab cycling shortcuts to be configurable in Editor

### LINKED ISSUE #4842: Add ability to change Ctrl+PgUp and PgDn shortcuts
I really dislike editors with MRU behaviour but at the same time I really like Ctrl+Tab and Ctrl+Shift+Tab because I don't have to remove my hand from the mouse like with Ctrl+PgUp or PgDn. I wish I had the ability to customize Ctrl+PgUp and PgDn just like Tab and Shift+Tab so I could replace them and use my preferred behaviour with my preferred shortcuts.

Thanks in advance

### PR DESCRIPTION
**Pull request by csabella**
Fixes #4842.  

- Added shortcuts to cycle through files.
- If statement to convert `Backtab` to `Shift+Tab` was only for 'nt', but problem existed on linux too, so removed the conditional.
- Tab navigation routine already existed in `Tabs` class, so moved it to `BaseTabs` class for it to be used in the editor (which creates a `BaseTabs` widget) without affecting existing code (like in iPython console).

### CODE REVIEW

File: spyder/widgets/tabs.py [hunk 1]
```diff
@@ -385,6 +385,16 @@ def keyPressEvent(self, event):
                 handled = True
         if not handled:
             QTabWidget.keyPressEvent(self, event)
+
+    def tab_navigate(self, delta=1):
```
Comment (ccordoba12) [inline spyder/widgets/tabs.py]: @csabella, `Ctrl+Tab` is handled differently in the Editor, because there we show a dialog to select among the current opened files. So I think you need to revert this change.

Comment (csabella) [reply spyder/widgets/tabs.py]: @ccordoba12   It still works correctly in the Editor even with moving this.  Without changing preferences, `Ctrl+Tab` brings up the dialog to select the file in MRU  order and `Ctrl+PgDown` cycles through the files.  But, this code allows those keys to be configurable.

Comment (ccordoba12) [reply spyder/widgets/tabs.py]: O, great! Then this is ready.
